packages/mongodb-rag-ingest/src/sources/CodeOnGithubTextDataSource.ts

```diff
@@ -0,0 +1,88 @@
+import { Page, PageMetadata } from "mongodb-rag-core";
+import {
+  MakeGitHubDataSourceArgs,
+  makeGitHubDataSource,
+} from "./GitHubDataSource";
+import path from "path";
+
+export type MakeCodeOnGithubTextDataSourceParams =
+  // MakeGitHubDataSourceArgs & {
+  Omit<MakeGitHubDataSourceArgs, "handleDocumentInRepo"> & {
+    /**
+    Metadata to include with all Pages in DB.
+   */
+    metadata?: PageMetadata;
+  };
+
+/**
+  Loads source code files from a GitHub repo.
+ */
+export const makeCodeOnGithubTextDataSource = async ({
+  name,
+  repoUrl,
+  repoLoaderOptions,
+  filter,
+  metadata,
+}: MakeCodeOnGithubTextDataSourceParams) => {
+  return makeGitHubDataSource({
+    name,
+    repoUrl,
+    filter,
+    repoLoaderOptions: {
+      ...(repoLoaderOptions ?? {}),
+      ignoreFiles: [
+        /LICENSE/,
+        /CONTRIBUTING/,
+        /\.git/, // Ignores .git/, .gitignore, .github/, etc.
+        /\.dockerignore/,
+        /\.gcloudignore/,
+        /\.editorconfig/,
+        /\.vscode/,
+        ...(repoLoaderOptions?.ignoreFiles ?? []),
+      ],
+    },
+    async handleDocumentInRepo(document) {
+      const page: Page = {
+        body: document.pageContent,
+        format: "txt",
```

Comment:
why not let this be passable to account for different formats of different languages? 

`java`, `js`, etc

Reply:
supporting "intelligent" chunking on a variety of different languages shouldn't be too hard using the langchain `RecursiveCharacterTextSplitter.fromLanguage()` method.

Reply:
I went a little crazy on this one and added some machinery around a list of "canonical" page formats (and their synonyms). It's a superset of the existing type and should be relatively flexible, though maybe I should add a way to append to it from the ingest config 🤔